tests/sepp/Execution.cpp

```diff
@@ -91,22 +120,26 @@ Report Execution::run() {
     // TODO - record samples (memory usage, rocksdb stats, ...)
   }
 
-  _state.store(ExecutionState::kStopped);
+  // set our execution state to stopped
+  stop();
 
   waitUntilAllThreadsAre(ThreadState::kFinished);
 
   std::chrono::duration<double, std::milli> runtime =
       std::chrono::high_resolution_clock::now() - start;
-  return buildReport(runtime.count());
-}
 
-Report Execution::buildReport(double runtime) {
-  for (auto& thread : _threads) {
-    if (thread->_thread.joinable()) {
-      thread->_thread.join();
+  joinThreads();
```

Comment:
I don't think this is really necessary since after `waitUntilAllThreadsAre(ThreadState::kFinished)` all thread objects are fully updated so we can already access e.g. failure information even if the thread has not yet terminated. But if you prefer we can also leave it here.

Reply:
I think you are right, and it is not strictly necessary to call this. However, I left this line in to make it obvious to the reader that no threads will be dangling around here.